cmd/main.go

```diff
@@ -242,8 +244,73 @@ func main() {
 		}
 	}
 
-	nimState := os.Getenv("NIM_STATE")
 	signalHandlerCtx := ctrl.SetupSignalHandler()
+	setupNim(mgr, signalHandlerCtx, kubeClient)
+
+	// nolint:goconst
+	if os.Getenv(enableWebhooksEnv) != "false" {
+		if kserveWithMeshEnabled {
+			if err = webhookservingv1.SetupServiceWebhookWithManager(mgr); err != nil {
+				setupLog.Error(err, "unable to create webhook", "webhook", "Knative Service")
+				os.Exit(1)
+			}
+		} else {
+			setupLog.Info("Skipping setup of Knative Service validating/mutating Webhook, " +
+				"because KServe Serverless setup seems to be disabled.")
+		}
+	}
+	// nolint:goconst
+	if os.Getenv(enableWebhooksEnv) != "false" {
+		if err = webhookservingv1beta1.SetupInferenceServiceWebhookWithManager(mgr); err != nil {
+			setupLog.Error(err, "unable to create webhook", "webhook", "InferenceService")
+			os.Exit(1)
+		}
+	}
+	// nolint:goconst
+	if os.Getenv(enableWebhooksEnv) != "false" {
```

Comment:
Do we need to put this into #284 line? 

if there is no iIG crd, why we add this webhook? However, I think we need to add this because of webhook manifests. So IG webhook manifests will be created anyway. Instead, we should check  if there is IG crd on cluster or not from webhook. wdyt?

Reply:
I don't think the code of the webhook will be invoked at all if the CRD is missing. In any case, there is a type assertion at the very begging of the code of the webhook.

Reply:
then how to manage manifest for the IG webhook? the manifests would be created anyway. am I right?